Basic Lobstrosity

Jacquerel opened the pull request:
## About The Pull Request

I'm slowly chipping away at mining mobs. These ones also got some new sprites because the old ones were a bit weird except when facing South.
![image](https://github.com/tgstation/tgstation/assets/7483112/015b8819-bab8-471a-86ae-70b1597ae327)
Arctic Lobstrosities are now hairy to give them a little more visual distinction from Lavaland ones.

In terms of behaviour, they're now a little faster and can charge you from further away.
They will _only_ attack players who are incapacitated in some way (primarily from being hit by their charge, but could be from a Goliath or something too) and will otherwise keep their distance until they can charge again. They move slower for a short duration after charging though, so you have time to slap them a bit.

If a Lobstrosity downs you then it will try to snip off one of your arms, then retreat in order to eat it.
Obviously nobody likes losing an arm, but this does give you an opportunity to get away while it is distracted? Funnily enough the way our health system works means that sometimes losing that arm actually takes you out of soft crit so you can stumble back to the station for a replacement (or try to wrestle yours back?)

All of these things are achievable also by a player if you make one sapient, they will pull arms off mobs they attack which are in crit and can eat arms if they see them lying around if they want.
I added an element to let you dismember people with your bare hands, maybe someone evil can use it to add a beheading attack some day.

Here's a video of their new behaviours:
https://www.youtube.com/watch?v=9eKxsH7hD7Q

## Why It's Good For The Game

Gives mobs more character.
Reduces our list of frozen simple mobs.
Replaces some ugly side sprites.
Medbay enrichment?

## Changelog

:cl:
refactor: Lobstrosities are now basic mobs and have different AI behaviour. Please report anything which seems like it shouldn't be happening.
add: Lobstrosities will now only opportunistically attack things they have knocked over with their charge, and are otherwise timid.
add: Lobstrosities are hungry for fingers and will steal one of your arms if they defeat you in combat, although this gives you time to crawl away.
sprite: New sprites for Lobstrosities.
/:cl:

Jacquerel committed:
New sprites! Cool!

Jacquerel committed:
Merge branch 'master' into dod-a-chok-dud-a-chum

Jacquerel committed:
Hit & Run

Jacquerel committed:
apply to correct subtype

Jacquerel committed:
This fucks with it

Jacquerel committed:
I think this is about how I want it now

Jacquerel committed:
Hehe pinchy

Jacquerel committed:
your fingers are mine

Jacquerel committed:
Begone

Jacquerel committed:
Rename script

Jacquerel committed:
Forgot my file extension, still worked though

Jacquerel committed:
Weird newline added by script

GuillaumePrata commented:
> Obviously nobody likes losing an arm, but this does give you an opportunity to get away while it is distracted? Funnily enough the way our health system works means that sometimes losing that arm actually takes you out of soft crit

Not only is this behavior funny, but it also "removes" the worst part of our mining desing.
Dying as miner is overly punishing (Ignoring the round removals) with how long it takes for you to be found and revived. Now if you are lucky it is replaced with a fair punishment, losing a limb, but you can just walk back to the station yourself and get it fixed while cutting on the annoying "wait and hope someone comes help me" period of failures as a miner.

This honestly should be a design goal moving forward whenever it makes sense.

Jacquerel committed:
Go away

Jacquerel committed:
Why won't you revert

Jacquerel committed:
There we go

Jacquerel committed:
Don't make a bunch of identical elements

carlarctg commented:
> > Obviously nobody likes losing an arm, but this does give you an opportunity to get away while it is distracted? Funnily enough the way our health system works means that sometimes losing that arm actually takes you out of soft crit
> 
> Not only is this behavior funny, but it also "removes" the worst part of our mining desing. Dying as miner is overly punishing (Ignoring the round removals) with how long it takes for you to be found and revived. Now if you are lucky it is replaced with a fair punishment, losing a limb, but you can just walk back to the station yourself and get it fixed while cutting on the annoying "wait and hope someone comes help me" period of failures as a miner.
> 
> This honestly should be a design goal moving forward whenever it makes sense.

Agreed, it makes me wonder of similar replacements for legions, watchers, goliaths.

atlasle commented:
> Agreed, it makes me wonder of similar replacements for legions, watchers, goliaths.
Goliaths need like lairs that they pull you to, that would also home some loot (maybe like goliath eggs or ambergris) that would reward other miners for saving you (also we really need to be able to attach a health sensor to a GPS to activate a distress beacon that would make the signal blink red or something)

SimplySmiley123 commented:
> > Agreed, it makes me wonder of similar replacements for legions, watchers, goliaths.
> 
> Goliaths need like lairs that they pull you to, that would also home some loot (maybe like goliath eggs or ambergris) that would reward other miners for saving you (also we really need to be able to attach a health sensor to a GPS to activate a distress beacon that would make the signal blink red or something)

This would honestly be an amazing thing to have.
Especially when it comes to miners who forget to name their GPS something to let the other miners or the singular paramedic that gives a damn which GPS signal to look for.

Jacquerel commented:
I honestly don't have clear ideas for equivalents for any other mob but I will try to keep it in mind, it's an interesting concept.

san7890 commented on code/datums/ai/basic_mobs/basic_ai_behaviors/run_away_from_target.dm:
```diff
@@ -16,6 +18,8 @@
 
 /datum/ai_behavior/run_away_from_target/perform(seconds_per_tick, datum/ai_controller/controller, target_key, hiding_location_key)
 	. = ..()
+	if (!controller.blackboard[BB_BASIC_MOB_FLEEING])
```
i forgot, is fleeing a boolean or does it resolve to the mob it's fleeing from? use `isnull()` if it's the latter but i could just be smallbrained rn

Jacquerel replied:
It's just a boolean, don't worry I was thinking of you.

san7890 commented on code/datums/mapgen/Cavegens/IcemoonCaves.dm:
```diff
@@ -6,7 +6,7 @@
 						  /mob/living/simple_animal/hostile/asteroid/polarbear = 30, /obj/structure/spawner/ice_moon/polarbear = 3, \
 						  /mob/living/simple_animal/hostile/asteroid/hivelord/legion/snow = 50, /mob/living/simple_animal/hostile/asteroid/goldgrub = 10, \
-						  /mob/living/simple_animal/hostile/asteroid/lobstrosity = 15)
+						  /mob/living/basic/mining/lobstrosity = 15)
```
A) alphabetize
B) this list sucks can you properly multiline it

san7890 commented on code/modules/fishing/sources/source_types.dm:
```diff
@@ -71,7 +71,7 @@
 		FISHING_DUD = 5,
 		/obj/item/fish/chasm_crab/ice = 15,
-		/mob/living/simple_animal/hostile/asteroid/lobstrosity = 1,
+		/mob/living/basic/mining/lobstrosity = 1,
```
alphabetize

san7890 commented on code/modules/fishing/fish/chasm_detritus.dm:
```diff
@@ -22,16 +22,16 @@
-			/mob/living/simple_animal/hostile/asteroid/lobstrosity/lava = 1,
+			/mob/living/basic/mining/lobstrosity/lava = 1,
 		),
 		BODIES_ONLY = list(
 			/obj/effect/mob_spawn/corpse/human/skeleton = 3,
-			/mob/living/simple_animal/hostile/asteroid/lobstrosity/lava = 1,
+			/mob/living/basic/mining/lobstrosity/lava = 1,
 		),
 		NO_CORPSES = list(
 			/obj/item/stack/ore/slag = 10,
 			/obj/item/stack/sheet/bone = 14,
-			/mob/living/simple_animal/hostile/asteroid/lobstrosity/lava = 1,
+			/mob/living/basic/mining/lobstrosity/lava = 1,
```
alphabetize all three of these lists

san7890 commented on code/modules/mob/living/basic/lavaland/goliath/goliath_trophy.dm:
```diff
@@ -0,0 +1,19 @@
+	icon_state = "goliath_tentacle"
+	denied_type = /obj/item/crusher_trophy/goliath_tentacle
+	bonus_value = 2
+	var/missing_health_ratio = 0.1
+	var/missing_health_desc = 10
```
would you mind dmdoccing while we're here

san7890 commented on code/modules/mob/living/basic/lavaland/lobstrosity/lobstrosity.dm:
```diff
@@ -0,0 +1,86 @@
+	icon_dead = "lobstrosity_dead"
+
+/// Charge a long way, knock down for longer, and perform an instant melee attack
+/datum/action/cooldown/mob_cooldown/charge/basic_charge/lobster
```
lober

san7890 replied:
no change

Jacquerel committed:
Reorders lists mostly
Set goliath tentacle description programmatically because it was literally just a lie

Jacquerel commented:
Some of those lists I ordered by probability _and_ alphabetically instead based on my whim, vibes-based sorting.
I looked at the ai defines file, couldn't decide how to split it, and decided not to do it.

LSommers commented:
> Goliaths need like lairs that they pull you to, that would also home some loot (maybe like goliath eggs or ambergris) that would reward other miners for saving you (also we really need to be able to attach a health sensor to a GPS to activate a distress beacon that would make the signal blink red or something)

goliath egg that you can hatch into a useless but very cute baby goliath pet
(several assistants rush to lavaland and immediately die trying to get the new cosmetic item)

atlasle commented:
> goliath egg that you can hatch into a useless but very cute baby goliath pet
> 
> (several assistants rush to lavaland and immediately die trying to get the new cosmetic item)

Our brains are on the same wavelength
